snapcraft/tests/test_common.py

```diff
@@ -112,8 +111,13 @@ class ArchTestCase(testtools.TestCase):
 
     def setUp(self):
         super().setUp()
-        common.host_machine = platform.machine()
-        common.target_machine = common.host_machine
+        # tests will override host_machine, protect it first
+        ph = patch('snapcraft.common.host_machine', new=common.host_machine)
+        ph.start()
+        self.addCleanup(ph.stop)
+        pt = patch('snapcraft.common.target_machine', new=common.host_machine)
+        pt.start()
+        self.addCleanup(pt.stop)
```

Comment:
My searching on common patterns for these singletons (antipatterns really) showed that this is the most accepted way of doing it.

Ideally I want it to be just one class though and not many disparate attributes